lib/src/replay/network_performance/dio_interceptor.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:dio/dio.dart';
+import 'package:posthog_flutter/posthog_flutter.dart';
+
+class PosthogDioInterceptor extends Interceptor {
+  @override
+  Future<void> onResponse(
+    Response<dynamic> response,
+    ResponseInterceptorHandler handler,
+  ) async {
+    await _captureNetworkEvent(
+      response: response,
+    );
+    super.onResponse(response, handler);
+  }
+
+  @override
+  Future<void> onError(
+    DioException err,
+    ErrorInterceptorHandler handler,
+  ) async {
+    final Response<dynamic>? response = err.response;
+    if (response != null) {
+      await _captureNetworkEvent(response: response);
+    }
+    super.onError(err, handler);
+  }
+
+  Future<void> _captureNetworkEvent({
+    required Response<dynamic> response,
+  }) async {
+    final String url = response.requestOptions.uri.toString();
+    final String method = response.requestOptions.method;
+    final int statusCode = response.statusCode ?? 0;
+    final Object? publishableRequest = _tryTransformDataToPublishableObject(
+      data: response.requestOptions.data,
+    );
+    final Object? publishableResponse = _tryTransformDataToPublishableObject(
+      data: response.data,
+    );
+    final Map<String, Object> snapshotData = <String, Object>{
+      'type': 6,
+      'data': <String, Object>{
+        'plugin': 'rrweb/network@1',
+        'payload': <String, Object>{
+          'url': url,
+          'method': method,
+          'status_code': statusCode,
+          if (publishableRequest != null) 'request': publishableRequest,
+          if (publishableResponse != null) 'response': publishableResponse,
```

Comment:
i think we should skip this for now, or we should check its body size and type before appending to the properties.
events that are large will be dropped, and it has to be json serializable, so better to skip for a 1st version

Reply:
> Size concern: May I know what's the size limit here? If you let me know, I can drop it.

lets dow hat the JS SDK does
https://github.com/PostHog/posthog-js/blob/962649dcc87ff67da2e80e7dfb776020b903bc7b/packages/browser/src/extensions/replay/external/config.ts#L130-L170
you can inspire from this implementation